game-headed/src/main/java/org/triplea/game/client/HeadedGameRunner.java

```diff
@@ -66,6 +67,15 @@ public static void main(final String[] args) {
             SwingUtilities.invokeLater(
                 () -> DownloadMapsWindow.showDownloadMapsWindowAndDownload(mapName));
           });
+      MacOsIntegration.setOpenFilesHandler(
+          list ->
```

Comment:
Why is this a list?
I'm noting that as picking the first element we find could cause ordering dependencies or other incorrect behavior. Is this to cover if a person selects many files and opens them all?

Reply:
Would it make for a cleaner API if  `MacOsIntegration.setOpenFilesHandler(` were to take a `File`

Reply:
2nd question, is this only going to work if TripleA is already open? It's a bit odd to register a file handler for opening files? Isn't the idea that opening save game files would open TripleA and TripleA would not already need to be open.

Reply:
> Why is this a list?

No idea, it's probably for the case of many files opened.

> Would it make for a cleaner API if  `MacOsIntegration.setOpenFilesHandler(` were to take a `File`

Simplifying this might be an option, but currently this class is just a generic wrapper for the desktop api, all of the other methods don't really try to hide details either.

Aand no,this will work even if triplea is not opened.
See, on Linux and Windows file handling causes the os to launch a new process for every single file. Macos on the other hand doesn't really like having multiple processes, instead it prefers a single one with multiple windows.
So when triplea isn't opened already it will launch an instance and instantly send a file open event, similar to uri handling, and if another file is opened, the same process receives another event, thinking about it, this might have unforseen consequences when this happens during a game, besides the fact that there might be a race condition here anyways (the event might be fired before the main menu is instantiated

Reply:
Wait, we had our own custom logic in MacOS integration, it denigrated to just a wrapper, and now because of that we can't use it to hide details and it just needs to be a pass-thru wrapper?

- IMO the wrapper should try to hide details or it's better to add deprecation tags and not use it.
- Some commentary would be good to avoid confusion on how the file handles works, I don't know if it'll be clear to future maintainers how this works.

Reply:
> all of the other methods don't really try to hide details either.

They do not have similar details to hide and we've already discussed it probably makes sense to inline them.

The API added does not make sense and is not well documented. Should we really just pick a file out of many, or perhaps open no files and make it up to the user to pick the right one? A List API where many elements are ignored and only one is active, or allowing for a no-op if an empty list is not terribly well defined, the behavior is very non-deterministic.

Reply:
@DanVanAtta I simplified the API, let me know what you think.

Reply:
Much simpler, thank you!